external/dockerfile_functions.sh

```diff
@@ -500,7 +518,22 @@ generate_dockerfile() {
     build=$7
     testtarget=$8
 
-    set_test_info ${test}
+    check_external_custom_test=0
+
+    if [ ${test} == 'external_custom' ]; then
+        check_external_custom_test=1
+        echo "EXTERNAL_CUSTOM_REPO points to ${EXTERNAL_CUSTOM_REPO} in dockerfile_functions.sh"
+        echo "EXTERNAL_CUSTOM_BRANCH points to ${EXTERNAL_REPO_BRANCH} in dockerfile_functions.sh"
+        test="$(echo ${EXTERNAL_CUSTOM_REPO} | awk -F'/' '{print $NF}' | sed 's/.git//g')"
+        echo ${test}
+        tag_version=${EXTERNAL_REPO_BRANCH}
+    fi
+
+    if [[ ${check_external_custom_test} -eq 1 ]]; then
```

Comment:
We may  just move condition logic between Line 524 to Line 529  to line 532 using the same condition ` if [ ${test} == 'external_custom' ];` no need for the extra flag check_external_custom_test.

Reply:
I used the flag check_external_custom_test because in the line 527 we are changing the test to point to individual tests like netty, zookeeper instead of external_custom, so now the test in all the subsequent function calls, test no longer refers to external_custom.  So if we are to pass the external_custom_parameters to Dockerfile we need a check whether it comes from an external_custom_target eligibe test or not.